vpplink/interfaces.go

```diff
@@ -225,24 +212,87 @@ func (v *VppLink) AddInterfaceAddress(swIfIndex uint32, addr *net.IPNet) error {
 	return v.addDelInterfaceAddress(swIfIndex, addr, true)
 }
 
-func (v *VppLink) enableDisableInterfaceIP6(swIfIndex uint32, enable bool) error {
+func (v *VppLink) setUnsetInterfaceTag(swIfIndex uint32, tag string, isAdd bool) error {
 	v.lock.Lock()
 	defer v.lock.Unlock()
 
-	request := &vppip.SwInterfaceIP6EnableDisable{
+	request := &interfaces.SwInterfaceTagAddDel{
+		IsAdd:     isAdd,
 		SwIfIndex: interface_types.InterfaceIndex(swIfIndex),
-		Enable:    enable,
+		Tag:       tag,
+	}
+	response := &interfaces.SwInterfaceTagAddDelReply{}
+	err := v.ch.SendRequest(request).ReceiveReply(response)
+	if err != nil || response.Retval != 0 {
+		return fmt.Errorf("cannot add interface tag: %v %d", err, response.Retval)
 	}
+	return nil
+}
+
+func (v *VppLink) SetInterfaceTag(swIfIndex uint32, tag string) error {
+	return v.setUnsetInterfaceTag(swIfIndex, tag, true /* isAdd */)
+}
+
+func (v *VppLink) UnsetInterfaceTag(swIfIndex uint32, tag string) error {
+	return v.setUnsetInterfaceTag(swIfIndex, tag, false /* isAdd */)
+}
+
+func (v *VppLink) enableDisableInterfaceIP(swIfIndex uint32, isIP6 bool, isEnable bool) error {
+	v.lock.Lock()
+	defer v.lock.Unlock()
 	response := &vppip.SwInterfaceIP6EnableDisableReply{}
-	return v.ch.SendRequest(request).ReceiveReply(response)
+	request := &vppip.SwInterfaceIP6EnableDisable{
+		Enable:    isEnable,
+		SwIfIndex: interface_types.InterfaceIndex(swIfIndex),
+		// Af: types.GetBoolIPFamily(isIP6),
```

Comment:
This looks weird, what is the behavior if no Af is specified? That's worth a comment at least imo :)

Reply:
Oh, right. Sorry about this,
This is a remainder of trying to move away from unnumbered interfaces, and using 
`SwInterfaceEnableDisable { Af : isIP6 }` but finally reverted to `SwInterfaceIP6EnableDisable`

Reply:
Ah right, missed the IP6 in the message name 😄